### ISSUE TITLE
Tensor tools conversion fails

### ISSUE THREAD

**Issue by guillaume-be** [LaurentMazare/tch-rs#154]
Hello,

I am trying to understand how to convert my existing Pytorch models (`.pth` format) to the` .ot` format used in this crate. I understand that saving the pickled `.pth` files to `.npz` would be the way to go as an intermediate step.

To better understand the format of the `npz` to generate, I tried converting the existing `.ot` files given in the examples back to `npz`. The following fails with the latest version of master:

```rust
    let src_filename = "E:/files/resnet18.ot";
    let dst_filename = "E:/files/resnet18.npz";
    let tensors = if src_filename.ends_with(".npz") {
        tch::Tensor::read_npz(src_filename)?
    } else if src_filename.ends_with(".ot") {
        tch::Tensor::load_multi(src_filename)?
    } else {
        bail!("unhandled file {}", src_filename)
    };
    for (name, tensor) in tensors.iter() {
        println!("{}: {} {:?}", src_filename, name, tensor)
    }
    if dst_filename.ends_with(".npz") {
        tch::Tensor::write_npz(&tensors, dst_filename)?
    } else if dst_filename.ends_with(".ot") {
        tch::Tensor::save_multi(&tensors, dst_filename)?
    } else {
        bail!("unhandled file {}", dst_filename)
    };
    Ok(())
```

with `Error: ErrorMessage { msg: "incoherent elt kind, Float != Uint8" }`

Am I missing something?

**Comment by LaurentMazare** [comment]
Thanks for reporting this issue, I just pushed a fix for it.
You're right that the normal way to port model weights to rust is to write them to a npz file from python and then load them from rust and create an ot file. Note that the issue you reported was only happening when writing a npz file so should not be involved in this process.

Here is the simple python script that I used to export the weights for various computer vision models. I would then run `cargo run -example tensor-tools cp resnet18.npz resnet18.ot`.
Let me know if you run into any issue porting your weights from python to rust.

```python
import numpy as np
import torch
import torchvision

m = torchvision.models.resnet18(pretrained=True)
nps = {}
for k, v in m.state_dict().items(): nps[k] = v.numpy()
np.savez('resnet18.npz', **nps)
```
